spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -110,12 +113,14 @@ else if (binderConfigurations.size() == 1) {
 				defaultDomain += ".";
 			}
 			args.add("--spring.jmx.default-domain=" + defaultDomain + "binder." + configurationName);
-			args.add("--spring.application.admin.enabled=false");
 			SpringApplicationBuilder springApplicationBuilder =
 					new SpringApplicationBuilder()
 							.sources(binderConfiguration.getBinderType().getConfigurationClasses())
 							.bannerMode(Mode.OFF)
 							.web(false);
+			if (binderProperties.isEmpty() && context != null) {
```

Comment:
Might be worth a brief comment here to explain why we only set the parent context in this case.